components/public-api/typescript-common/src/public-api-converter.ts

```diff
@@ -1046,6 +1046,10 @@ export class PublicAPIConverter {
             pinnedEditorVersions: settings.pinnedEditorVersions || {},
             restrictedEditorNames: settings.restrictedEditorNames || [],
             defaultRole: settings.defaultRole || undefined,
+            timeoutSettings: {
+                inactivity: settings.timeoutSettings?.inactivity ? this.toDuration(settings.timeoutSettings?.inactivity) : undefined,
+                allowChangeByMembers: settings.timeoutSettings?.allowChangeByMembers,
```

Comment:
How does this work if the definition is 
```
export type TimeoutSettings = {
    inactivity?: string;
    allowChangeByMembers: boolean;
};
```
? :thinking:

I'm looking a bit closer here, because I saw the following:
 1. right after org creation/upgrade, I tried to set "inactivity" to `31m`
 2. hitting save made everything flicker. On reload, the `31m` was gone. The result from the network request was: `timeoutSettings: {}`
 3. then I explicitly toggled the "allowChangeByMembers" (it showed true before, I toggled it twice)
 4. hitting save now works.

Reply:
Also, the timeoutSettings shape now looks like this for all existing orgs:
```
timeoutSettings: {
    inactivity: undefined,
    allowChangeByMembers: undefined,
}
```
It should be `allowChangeByMembers: true` (and we can get rid of the "true" default in the UI).

Also, I'd argue we should push this "default" behavior into the Service ([link](https://github.com/gitpod-io/gitpod/blob/131a03373167d758b2cc39b390365c6f4412f0c6/components/server/src/orgs/organization-service.ts#L419)).

Reply:
Agreed about having the `true` default for allowing members to change the timeouts, will push that. Both properties on `TimeoutSettings` should be optional, I also will change that.